### PR TITLE
feat: add hot and archive db repositories for blobSidecars

### PR DESCRIPTION
**Pull request by g11tech**
Furthering the integration for
 - https://github.com/ChainSafe/lodestar/pull/5181
 
this PR add dbs for blobsidecars which opens the way for integrating network changes (req/resp first and then gossip)
The old dbs related to blobs will be cleaned up post migration is complete

### CODE REVIEW

**Review by dapplion — APPROVED** [review]

File: packages/beacon-node/src/api/impl/beacon/blocks/index.ts [hunk 1]
```diff
@@ -265,9 +265,22 @@ export function getBeaconBlockApi({
...
+      const {block, executionOptimistic} = await resolveBlockId(chain.forkChoice, db, blockId);
+      const blockRoot = config.getForkTypes(block.message.slot).BeaconBlock.hashTreeRoot(block.message);
+
+      let {blobSidecars} = (await db.blobSidecars.get(blockRoot)) ?? {};
+      if (!blobSidecars) {
+        ({blobSidecars} = (await db.blobSidecarsArchive.get(block.message.slot)) ?? {});
+      }
+
+      if (!blobSidecars) {
+        throw Error("Not found in db");
```
Comment (twoeths) [inline packages/beacon-node/src/api/impl/beacon/blocks/index.ts:278]: we may want to add block slot/root to error message to help debug later

File: packages/beacon-node/src/db/repositories/blobSidecarsArchive.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+import {Bucket, Db, Repository} from "@lodestar/db";
+import {Slot} from "@lodestar/types";
+import {bytesToInt} from "@lodestar/utils";
+
+import {blobSidecarsWrapperSsz, BlobSidecarsWrapper} from "./blobSidecars.js";
+
+/**
+ * blobSidecarsWrapper by slot
+ *
+ * Used to store unfinalized BlobsSidecar
```
Comment (twoeths) [inline packages/beacon-node/src/db/repositories/blobSidecarsArchive.ts:11]:
```suggestion
 * Used to store finalized BlobsSidecar
```

**Review by twoeths — APPROVED** [review]
@g11tech  I dropped 2 minor points, could be fixed in upcoming PRs

**Comment by g11tech** [comment]
> @g11tech I dropped 2 minor points, could be fixed in upcoming PRs

sounds good, thanks will update

**Comment by wemeetagain** [comment]
:tada: This PR is included in [v1.9.0](https://github.com/ChainSafe/lodestar/releases/tag/v1.9.0) :tada:

**Pull request merged**